Lib/test/test_asyncio/test_base_events.py

```diff
@@ -2214,17 +2214,48 @@ def check_set_nodelay(self, sock):
     @unittest.skipUnless(hasattr(socket, 'TCP_NODELAY'),
                          'need socket.TCP_NODELAY')
     def test_set_nodelay(self):
-        sock = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM,
-                             proto=socket.IPPROTO_TCP)
-        with sock:
-            self.check_set_nodelay(sock)
+        with self.subTest('non-blocking'):
+            sock = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM,
+                                 proto=socket.IPPROTO_TCP)
+            with sock:
+                self.check_set_nodelay(sock)
+
+        with self.subTest('blocking'):
+            sock = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM,
+                                 proto=socket.IPPROTO_TCP)
+            with sock:
+                sock.setblocking(False)
+                self.check_set_nodelay(sock)
+
+
+    def check_set_quickack(self, sock):
+        # quickack already true by default on some OSes
+        opt = sock.getsockopt(socket.IPPROTO_TCP, socket.TCP_QUICKACK)
+        if opt:
+            base_events._set_quickack(sock, 0)
+
+        opt = sock.getsockopt(socket.IPPROTO_TCP, socket.TCP_QUICKACK)
+        self.assertFalse(opt)
 
-        sock = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM,
-                             proto=socket.IPPROTO_TCP)
-        with sock:
-            sock.setblocking(False)
-            self.check_set_nodelay(sock)
+        base_events._set_quickack(sock, 1)
+
+        opt = sock.getsockopt(socket.IPPROTO_TCP, socket.TCP_QUICKACK)
+        self.assertTrue(opt)
 
+    @unittest.skipUnless(hasattr(socket, 'TCP_QUICKACK'),
+                         'need socket.TCP_QUICKACK')
+    def test_set_quickack(self):
```

Comment:
Would it be possible to move this test in test_socket?

Reply:
Would you like me to move the rest of the socket related tests as well?  Seems like more of a refactor.

Reply:
This test has nothing specific to asyncio, and quick ACK is different: it uses a dedicated private member (quick_ack) on Windows. I would prefer to have a test in test_socket.

Reply:
didn't realize nodelay had consumers here (why it's here) when I copied it, will move